src/Compilers/Core/MSBuildTask/MvidReader.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public sealed class MvidReader : BinaryReader
+    {
+        public MvidReader(Stream stream) : base(stream)
+        {
+        }
+
+        public static Guid ReadAssemblyMvid(Stream stream)
+        {
+            try
+            {
+                var mvidReader = new MvidReader(stream);
+                return mvidReader.FindMvid();
+            }
+            catch (EndOfStreamException)
+            {
+            }
+
+            return Guid.Empty;
```

Comment:
I think so. If the ref assembly is truncated (there are section headers, but no actual sections), we risk reading past the end of the file.